### ISSUE TITLE
[Feature Request]: Transient "hide watched" toggle

### ISSUE THREAD

**Issue by jasonhenriquez** [FreeTubeApp/FreeTube#4999]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

As a user, I sometimes want to hide watched videos in my subscriptions only sometimes. Rather than have to enable and disable a permanent setting, I would like to be able to do it on a whim and have it revert when it's no longer desired.

### Proposed Solution

Implement transient toggle switch somewhere on the Subscriptions page (like the `ft-refresh-widget` or the top of the page) that overrides the "Hide Videos on Watch" preference until the route is changed.

### Alternatives Considered

Have the toggle override the default setting for the session instead of just until route change.

### Issue Labels

ease of use improvement, improvement to existing feature

**Comment by MarmadileManteater** [comment]
My opinion is that we should try to limit the number of floating transient things on the page, but I understand the desire for this feature because the current `Hide Watched` setting is not reactive as-is.

**Comment by jasonhenriquez** [comment]
I agree with both of those sentiments. Part of my mindset is that I would never really want to have "Hide Videos on Watch" enabled permanently, but I would like to be able to toggle it on/off at will when looking at my open subscription videos. I sometimes will open a video and want to go back to it when I see it on my feed, so I wouldn't want it always be on. You could also make a similar but less salient case for "Only Show Latest Video for Each Channel". I wonder if something like an icon checkbox dropdown from #4231 would work for this.

**Comment by ChunkyProgrammer** [comment]
Maybe we could have a "Show Hidden results" link/button that appears if there's anything that's hidden (this can apply to subscriptions and searching if you have blocked any channels/tags). There should also probably be a parental control to prevent that link/button from showing though (on the search page at least)

**Comment by jasonhenriquez** [comment]
That's an interesting idea, I like that. Maybe that setting is a dropdown if we want to distinguish among "Hide 'Show hidden' button", "Always hide blocked words and channels", and "Default"? Or maybe just a toggle for MVP.

**Comment by jasonhenriquez** [comment]
Removing "good first issue" for now as the exact design is still somewhat in the air.

**Comment by Nobylo** [comment]
An alternative could be moving the entire settings category for the subscriptions page onto said page, under a drop-down menu. It wouldn't be transient which could be more inconvenient. I use the "show only latest video" toggle a lot in conjunction with hiding videos on watch, if one is gonna be on the subscriptions page for quick access then the other should probably be there as well for the same reasons. Fitting both could need a drop down menu and at that point you could shove the rest of the settings in there.

(something like this is what I'm thinking)
![FreeTube_nhAqisuBbm](https://github.com/user-attachments/assets/53a1fd55-3c73-41dd-8c49-4a873ca45aad)

**Comment by realrogue** [comment]
> An alternative could be moving the entire settings category for the subscriptions page onto said page, under a drop-down menu. It wouldn't be transient which could be more inconvenient. I use the "show only latest video" toggle a lot in conjunction with hiding videos on watch, if one is gonna be on the subscriptions page for quick access then the other should probably be there as well for the same reasons. Fitting both could need a drop down menu and at that point you could shove the rest of the settings in there.
> 
> (something like this is what I'm thinking) ![FreeTube_nhAqisuBbm](https://github.com/user-attachments/assets/53a1fd55-3c73-41dd-8c49-4a873ca45aad)

Oh this would be absolutely fantastic, are there any forks implementing this? I primarily want to be able to hide watched videos every now and again but the quick settings button idea and mockup is literally perfect.
